orangecontrib/spectroscopy/io/ptir.py

```diff
@@ -161,23 +201,42 @@ def read_spectra(self):
                 else:
                     intensities.append(data[0,:])
 
-        intensities = np.array(intensities)
+        spectra = np.array(intensities)
         features = np.array(wavenumbers)
         x_locs = np.array(x_locs).flatten()
         y_locs = np.array(y_locs).flatten()
+        z_locs = np.array(z_locs).flatten()
+
         if hyperspectra:
-            return _spectra_from_image(intensities, features, x_locs, y_locs)
-        else:
-            spectra = intensities
-
-            # locations
-            x_loc = y_loc = np.arange(spectra.shape[0])
-            metas = np.array([x_locs[x_loc], y_locs[y_loc]]).T
-
-            domain = Orange.data.Domain([], None,
-                                        metas=[Orange.data.ContinuousVariable.make(MAP_X_VAR),
-                                               Orange.data.ContinuousVariable.make(MAP_Y_VAR)]
-                                        )
-            data = Orange.data.Table.from_numpy(domain, X=np.zeros((len(spectra), 0)),
-                                                metas=np.asarray(metas, dtype=object))
-            return features, spectra, data
\ No newline at end of file
+            features, spectra, additional_table = _spectra_from_image(spectra, features, x_locs, y_locs)
+
+            new_attributes = []
+            new_columns = []
+            new_attributes.append(ContinuousVariable.make('z-focus'))
+            new_columns.append(np.full((len(z_locs),), z_locs))
+
+            domain = Domain(additional_table.domain.attributes,
+                            additional_table.domain.class_vars,
+                            additional_table.domain.metas + tuple(new_attributes))
+            data = additional_table.transform(domain)
+            with data.unlocked():
+                data[:, new_attributes] = np.asarray(new_columns).T
+            return features, spectra, data
+
+        # locations
+        x_loc = y_loc = z_loc = np.arange(spectra.shape[0])
```

Comment:
These three variables (`x_loc`, `y_loc`, `z_loc`) are all the same index array, and only used once. Call it `i_loc` or something and use that

Reply:
Actually, unless `x_locs` and friends can be longer than `spectra` (I don't know, can they?) why not just replace both of these lines with 
`metas = np.vstack((x_locs, y_locs, z_locs)).T`